torchrl/envs/transforms/transforms.py

```diff
@@ -4317,6 +4317,150 @@ def __repr__(self) -> str:
         )
 
 
+class Stack(Transform):
+    """Stacks tensors and tensordicts.
+
+    This transform is useful for environments that have multiple agents with
+    identical specs under different keys. The specs and tensordicts for the
+    agents can be stacked together under a shared key, in order to run MARL
+    algorithms that expect the tensors for observations, rewards, etc. to
+    contain batched data for all the agents.
+
+    Args:
+        in_keys (sequence of NestedKey): keys to be stacked. If `None` or not
+            provided, the keys will be retrieved from the group map of the
+            environment the first time the transform is used. This behavior will
+            only work if a parent is set.
+        out_key (NestedKey): key of the resulting tensor.
+        dim (int, optional): dimension to insert. Default is ``0``.
```

Comment:
there's usually some benefit in using negative dims: if your transform is being used by a single env, multiple envs (ie envs where batch-size is non-empty) and / or replay buffers, the `dim=0` arg will lead to different output shapes, putting a feature dimension before the batch size!
Not sure we want to strictly prevent that, but for instance in `FlattenObservation` we ask the user to explicitly tell us that they want the dim to be non-negative.

Reply:
I see, that makes sense. I can make the default `allow_positive_dim=False, dim=-1`.

It is worth pointing out that if the user wants to stack multi-agent data, the different tensors under the agents may have different numbers of dimensions, so they will have to set `allow_positive_dim=True, dim=0`. I'll add an example of that to the docstring

Reply:
Oh I didn't realize that if the elements to be stacked are tensordicts with `batch_size=[]`, `torch.stack(values, dim=-1)` is actually  equivalent to `torch.stack(values, dim=0)`, so the default of `allow_positive_dim=False, dim=-1` works fine